### ISSUE TITLE
first-timers-only: Add test to make sure result callbacks returns possible sentences

### ISSUE THREAD

**Issue by TalAter** [TalAter/annyang#170]
#### first-timers-only

This issue is tagged :octocat: **first-timers-only**. It is only for people who have never contributed to open source before, and are looking for an easy way take their first steps.

Thank you for your help :heart:

#### Background

Callbacks added using `annyang.addCallback('result', aFunction)` will be called after any speech is detected, whether a command was matched or not (see [docs](https://github.com/TalAter/annyang/tree/master/docs#addcallbacktype-callback-context)).

The callback function should be called with an array of the possible sentences the user may have said as the first argument.

#### The Issue

The test specs do not currently test to verify that the callback function is called with the possible sentences the user may have said

Add a test spec that verify this spec:
````javascript
  it('should call the callback with the the first argument containing an array of all possible Speech Recognition Alternatives the user may have said', func1);
````
This should be added into the following test suite:
````javascript
describe("annyang.addCallback('result')", function() {
````

The last test described inside `describe("annyang.addCallback('resultNoMatch')", function() {…});` in *test/spec/BasicSpec.js*, can be used as a template for adding the new spec described above.

#### Contributing to annyang

- [ ] Fork the repository from the [annyang GitHub page](https://github.com/TalAter/annyang).
- [ ] Clone a copy to your local machine with `$ git clone git@github.com:YOUR-GITHUB-USER-NAME/annyang.git`
- [ ] Make sure you have *node.js* and *npm* installed on your machine. You can use this [guide](https://docs.npmjs.com/getting-started/installing-node) for help.
- [ ] Install all of annyang's development dependencies with npm. `$ cd annyang; npm install`
- [ ] Run grunt to make sure everything runs smoothly `$ grunt`
- [ ] Code, code, code.
- [ ] Run `$ grunt` after making changes to verify that everything still works and run the build process. Alternatively, you can just run `$ grunt watch` once, and leave that process running. It will continuously run all the tests and build the files every time you make a change to one of annyang's files. It will even *beep* if you make an error, and help you debug it.:+1:
- [ ] Before committing your changes, the last step must always be running `$ grunt`. This makes sure everything works, and all files are kept up to date with your changes.
- [ ] Once you've made sure all your changes work correctly, and committed all your changes, push your local changes back to github with `$ git push -u origin master`
- [ ] Visit your fork on GitHub.com ([https://github.com/YOUR-USER-NAME/annyang](https://github.com/YOUR-USER-NAME/annyang)) and create a pull request for your changes.
- [ ] Makes sure your pull request describes exactly what you changed and references this issue (just include the issue number in the title like this: #49)

#### About the tests

annyang is tested using [Jasmine](http://jasmine.github.io/2.0/introduction.html).

BasicSpec.js contains a series of spec groups (e.g. `describe('a spec group', function() {})`) which each contain 1 or more specs (e.g. `it('should do stuff', function() {})`). Some of the spec groups also contain some code which runs before each spec (`beforeEach(function() {})`).

To simulate Speech Recognition in the testing environment, annyang uses a mock object called [Corti](https://github.com/TalAter/Corti) which mocks the browser's SpeechRecognition object. Corti also adds a number of utility functions to the SpeechRecognition object which simulate user actions (e.g. `say('Hello there')`), and allow checking the SpeechRecognition's status (e.g. `isListening() === true`).

**Comment by Charlotteis** [comment]
Another fantastic issue, going to post this on the @yourfirstpr twitter account tomorrow!

**Comment by TalAter** [comment]
Thank you @Charlotteis :blush:

**Comment by ZahraTee** [comment]
I'd be happy to take a look at this. :)

I'm just wondering whether there's a particular reason why "the" is repeated in the test spec outlined here? I notice it's the case for all of the test specs beginning "should call the callback with" in BasicSpec.js.

**Comment by TalAter** [comment]
Thanks @ZahraTee, I'm looking forward to your contribution.

Regarding the 'the', that was just a typo. I'm pushing a fix for that now. Nicely spotted :eyes:

**Comment by TalAter** [comment]
Closed in PR #171

Code looks great @ZahraTee.

Thank you for the help, and I hope it is the first of many :octocat: contributions to open source!

P.S. - When submitting pull requests, some project maintainers would ask you to submit your work as a single commit. In the future, you can squash multiple commits like the 2 you had in this pull request into a single commit before sending the pull request.
Check out this handy tutorial by @kentcdodds on how to do this - https://egghead.io/lessons/javascript-how-to-squash-multiple-git-commits?series=how-to-contribute-to-an-open-source-project-on-github

Thanks again :+1:
